### PR TITLE
feat: Add possibility to remove old content from trash

### PR DESCRIPTION
**Pull request by rdenarie**
Theses content could generate a referentialIntegrityException after a copy paste : Create a web content in 5.2.x or 5.3.x
Modify it then publish it few times
Copy paste the web content, 
Drop the initial content
When trying to remove it from trash, a ReferentialIntegrityException is thrown

This fix will allow to flush liveRevision in the pasted content, and so, delete the node in trash

### CODE REVIEW

**Comment by rdenarie** [comment]
@ahamdi this pr is a backport from the addons trashcleaner as the code is now in the product.

File: core/services/src/main/java/org/exoplatform/services/cms/documents/job/TrashCleanerJob.java [hunk 1]
```diff
@@ -146,8 +146,10 @@ public void deleteNode(Node node) throws Exception {
       nodeToDelete.getSession().save();
       LOG.debug("Node " + nodeToDelete.getPath() + " deleted");
     } catch (ReferentialIntegrityException ref) {
-      LOG.error("ReferentialIntegrityException when removing " + node.getName() + " node from Trash", ref);
-    } catch (ConstraintViolationException cons) {
+      if (!fixReferentialIntegrityException(node)) {
+        LOG.error("ReferentialIntegrityException when removing " + node.getName() + " node from Trash", ref);
+        //log nothing if the fix success to deleteNode
+      }    } catch (ConstraintViolationException cons) {
```
Comment (ahamdi) [inline core/services/src/main/java/org/exoplatform/services/cms/documents/job/TrashCleanerJob.java]:
```suggestion
      }
    } catch (ConstraintViolationException cons) {
```

File: core/services/src/main/java/org/exoplatform/services/cms/documents/job/TrashCleanerJob.java [hunk 2]
```diff
@@ -186,6 +189,67 @@ private boolean checkPermission(Node node, String permissionType) throws Reposit
...
+      session.refresh(false);
+      name=node.getName();
+      boolean shouldDeleteNode = false;
+
+      String liveRevision = node.getProperty("publication:liveRevision").getValue().getString();
+      Node versionHistory = session.getNodeByUUID(liveRevision).getParent();
+      for (NodeIterator it = versionHistory.getNodes(); it.hasNext(); ) {
+        Node child = it.nextNode();
+        long nbRef = child.getReferences().getSize();
+        LOG.info("Version history child node {} have {} references",child.getPath(),nbRef);
```
Comment (ahamdi) [inline core/services/src/main/java/org/exoplatform/services/cms/documents/job/TrashCleanerJob.java]: we can get rid of those debugging logs WDYT ?

Comment (rdenarie) [reply core/services/src/main/java/org/exoplatform/services/cms/documents/job/TrashCleanerJob.java]: done

**Comment by rdenarie** [comment]
cc @ahamdi can you validate ?
Thanks

**Review by ahamdi — APPROVED** [review]

**Pull request merged**